automated install/basic-install.sh

```diff
@@ -1639,8 +1639,20 @@ install_dependent_packages() {
         done
         if [[ "${#installArray[@]}" -gt 0 ]]; then
             test_dpkg_lock
-            debconf-apt-progress -- "${PKG_INSTALL[@]}" "${installArray[@]}"
-            return
```

Comment:
It works, but it would mean that in case whiptail and debconf are both not installed, one additional apt-get call is done, thus two apt-get calls without progress bar.

To avoid this, the following could be done:
```
if is_command debconf-apt-progress && dpkg-query -s whiptail &> /dev/null; then
    debconf-apt-progress -- "${PKG_INSTALL[@]}" "${installArray[@]}"
else
    "${PKG_INSTALL[@]}" "${installArray[@]}" &> /dev/null
fi
```
The above single whiptail check/install is then not required.
As well those two `return` are obsolete since `return 0` is just below? Or is it important to have the empty line only when no package is in the array?

Additionally I am not sure if supressing any output of `apt-get` is great. It means that there can be a quite long time where user sees no progress. And in case of error, the installer just exits without any hint?
I would show full output, but with the `-qq` apt-get option which shows least overall output and inherits `-y` as well.

Reply:
We're talking over just removing the apt progress overlay. It causes issues when there is a need for user action (if a new config file comes in and the user needs to review changes). With the progress bar overlay the install process just stops and drops back to the terminal prompt. I added the apt progress as eye candy basically way back when. It is most likely time for it to go away.

Reply:
The `installArray` has already been filtered via `dpkg-query` (see lines 1633-1637) so there should be no need to check for `whiptail` a second time.

A few of us have had some internal discussions and we are talking about removing `debconf-apt-progress`.  
If we remove `debconf-apt-progress` then this PR and the previous PR's you created will be closed and we will re-factor to simply use `apt-get` and still display progress via a method _TBD_.

Reply:
@dschaper @bcambl 
> The `installArray` has already been filtered via `dpkg-query` (see lines 1633-1637) so there should be no need to check for `whiptail` a second time.

Ah that's true. However as long as the `install_dependent_packages` function should be working regardless of how it is used, both, `debconf-apt-progress` binary and `whiptail` presence (or simply the binary... this is actually faster: `command -v whiptail &> /dev/null`) need to be checked. So e.g. some change is done in the feature to install some other package prior to `whiptail` for some reason, it would fall back to `apt-get`, while currently it only works if `whiptail` (when missing) is added to the very first package array.

The other point was that with the above, there can be three APT installs happen.
1. `whiptail` missing (added to array) => whiptail separately installed via `apt-get`
2. `debconf` missing => Whole installer deps array installed via `apt-get`
3. Remaining Pi-hole deps array installed via `debconf-apt-progress`

Would be a bid simpler to have?:
1. Either `whiptail` or `debconf` missing => Whole installer deps array installed via `apt-get`
2. Remaining Pi-hole deps array installed via `debconf-apt-progress`

However this is just minor fine tuning...
_________________
> With the progress bar overlay the install process just stops and drops back to the terminal prompt.

That is true. One could work around this by using the non-interactive frontend and desired force options. E.g. on DietPi we use:
```
DEBIAN_FRONTEND=noninteractive apt-get -qq --allow-change-held-packages -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" install <pkg>
```
- The two DPKG options via config file.
- This assures an automated install and always leaves old config files in place while the new ones are stored with `.dpkg-dist` suffix or similar.
- `-qq` as said produces minimal output, but still certain lines. However at least `STDERR` should be not muted to not exit without any hint for debugging.

Reply:
I'm not comfortable forcing a choice upon a user. There may be a need to update to a newer configuration due to security updates or fixes for breaking changes. The user needs to know that things have changed and there is a possibility that user intervention is needed.

The points are all moot though as we will just remove the progress bar.

Reply:
#1756 suggests a similar fix.

Reply:
@Mausy5043 
`eval` is not required here as you can run the commands directly, leaving variable expansion to the shell. However the result is the same, since `debconf-apt-progress` is skipped.

@dschaper 
Jep, for a software title installer I actually agree that the installer should leave those choices to user, especially since the Pi-hole installer is interactive anyway.
Optionally instead when using `--unattended` (not sure how well this is implemented anyway currently?), this choice could/should be made, but then the docs/help should contain the result. Keeping old configs (and having new configs stored beside with `.dpkg-dist`) is at least the safest/conservative default then. As long as one does not run "apt-get dist-upgrade" actually software titles from official repo _must_ always remain working/configs _must_ be backwards compatible, although especially on Raspbian, those rules probably are not taken that strict.

Reply:
@MichaIng I cannot remember why I used `eval` in the first place. I may have considered that essential. But, my memory leaves me.